Intersect.Server.Core/Maps/MapController.cs

```diff
@@ -415,16 +414,18 @@ public void RespawnAllInstances()
         /// </summary>
         /// <param name="instanceId"></param>
         /// <returns>Whether or not we needed to create a new map instance</returns>
-        public bool TryCreateInstance(Guid instanceId, out MapInstance newLayer)
+        public bool TryCreateInstance(Guid instanceId, out MapInstance newLayer, Player creator)
         {
             newLayer = null;
             lock (GetMapLock())
             {
                 if (!mInstances.ContainsKey(instanceId))
                 {
                     Log.Debug($"Creating new instance with ID {instanceId} for map {Name}");
-                    mInstances[instanceId] = new MapInstance(this, instanceId);
-                    mInstances[instanceId].Initialize();
+                    var newMapInstance = new MapInstance(this, instanceId, creator);
+                    newMapInstance.Initialize();
+                    mInstances[instanceId] = newMapInstance;
+
                     newLayer = mInstances[instanceId];
```

Comment:
Can you replace `var newMapInstance =` with `newLayer =` rather than adding a local variable to a dictionary and then doing a lookup for a local variable?